tools/code-owners-parser/CodeOwnersParser/MatchedCodeownersEntry.cs

```diff
@@ -136,21 +140,41 @@ private static bool ContainsUnsupportedCharacters(string codeownersPath)
 
         private static bool ContainsUnsupportedSequences(string codeownersPath)
         {
+            if (codeownersPath == "/")
+            {
+                // This behavior matches GitHub CODEOWNERS interpreter behavior.
+                // I.e. a path of just "/" is unsupported.
+                Console.Error.WriteLine(
+                    $"CODEOWNERS path \"{codeownersPath}\" will never match. " +
+                    "Use \"/**\" instead.");
+                return true;
+            }
+
+            // See comment below why we support this path.
+            if (codeownersPath == "/**")
+                return false;
+
             // We do not support suffix of "/**" because it is equivalent to "/".
-            // We do not support suffix of "/**/" because it is equivalent to
-            // "/**/**" which is equivalent to "/".
-            string[] unsupportedSuffixSentences = { "/**", "/**/" };
+            // For example, "/foo/**" is equivalent to "/foo/"
+            // One exception to this rule is if the entire path is "/**":
+            // GitHub doesn't match "/" to anything if it is the entire path,
+            // and instead expects "/**".
+            if (codeownersPath != "/**" && codeownersPath.EndsWith("/**"))
+            {
+                Console.Error.WriteLine(
+                    $"CODEOWNERS path \"{codeownersPath}\" ends with " +
+                    "unsupported sequence of \"/**\". Replace it with \"/\".");
+                return true;
+            }
 
-            foreach (var sequence in unsupportedSuffixSentences)
+            // We do not support suffix of "/**/" because it is equivalent to
+            // suffix "/**" which is equivalent to suffix "/".
+            if (codeownersPath.EndsWith("/**/"))
             {
-                if (codeownersPath.EndsWith(sequence))
-                {
-                    Console.Error.WriteLine(
-                        $"CODEOWNERS path \"{codeownersPath}\" ends with " +
-                        $"unsupported sequence of \"{sequence}\". Replace it with \"/\". " +
-                        "Until then this path will never match.");
-                    return true;
-                }                
+                Console.Error.WriteLine(
+                    $"CODEOWNERS path \"{codeownersPath}\" ends with " +
+                    "unsupported sequence of \"/**/\". Replace it with \"/**\".");
```

Comment:
```suggestion
                    "unsupported sequence of \"/**/\". Replace it with \"/\".");
```
Why not just suggest to end with `/`

Reply:
@weshaggard per [this comment](https://github.com/Azure/azure-sdk-tools/pull/5308#issuecomment-1413228282), this is a special case:

![image](https://user-images.githubusercontent.com/4429827/216435322-0729e565-7170-44f5-b169-b1a84ffe780e.png)

Reply:
I agree for that one edge case but what about a path like `/foo/bar/**/` if they change to `/foo/bar/**` then your other rule above will trigger on that and tell them to change to `/foo/bar`.

Reply:
> /foo/bar/**/

Yeah, I also realized that, and have since fixed it in:

- https://github.com/Azure/azure-sdk-tools/pull/5325

Thanks!